src/python/pants/option/options.py

```diff
@@ -459,14 +449,11 @@ def for_scope(
                         )
                     )
             setattr(builder, dest, RankedValue(rank, val))
-        native_values = builder.build()
 
         # Check for any deprecation conditions, which are evaluated using `self._flag_matchers`.
         if check_deprecations:
-            native_values_builder = native_values.to_builder()
-            self._check_and_apply_deprecations(scope, native_values_builder)
-            native_values = native_values_builder.build()
-        return native_values
+            self._check_and_apply_deprecations(scope, builder)
```

Comment:
This is just some local streamlining that could have been done sooner. We don't need to build an OptionsValueContainer and then call `to_builder()` on it, when we already have a builder in hand...